src/controllers/dynakube/connectioninfo/reconciler.go

```diff
@@ -15,6 +15,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+var NoOneAgentCommunicationHostsError = errors.New("no communication hosts for OneAgent are available")
```

Comment:
This is not the way of creating a special error and checking for it in another place,
this way any other package could just overwrite this `var` and totally break the whole logic. (Also the moment this error is wrapped in another error the `errors.Is` will break)

The proper way is to create a `type` that implements the `Error` interface ([see more](https://go.dev/blog/error-handling-and-go)), for which you can check for with `errors.As(err, MySpecialError)`. Which will return `true` or `false` and do the conversion aswell.

Reply:
Actually the doc of Is says: 
``` // Is reports whether any error in err's chain matches target. ```
So wrapping should not be a problem. But I can make it an own type. Wanted to keep things as simple as possible.

Reply:
I honestly think, it makes error type checking cumbersome because you need an extra var for no other reason. Probably will wrap this in an extra helper.

Reply:
discussed offline, outcome: this is fine